python/ray/_raylet.pyx

```diff
@@ -456,20 +465,34 @@ cdef prepare_args_internal(
                 unique_ptr[CTaskArg](new CTaskArgByReference(
                     c_arg,
                     c_owner_address,
-                    arg.call_site())))
+                    (<ObjectRef>arg).call_site_data)))  # Avoid calling Python function
 
         else:
-            try:
-                serialized_arg = worker.get_serialization_context(
-                ).serialize(arg)
-            except TypeError as e:
-                msg = (
-                    "Could not serialize the argument "
-                    f"{repr(arg)} for a task or actor "
-                    f"{function_descriptor.repr}. Check "
-                    "https://docs.ray.io/en/master/ray-core/objects/serialization.html#troubleshooting " # noqa
-                    "for more information.")
-                raise TypeError(msg) from e
+            # The type check is because some custom types may not implement __eq__
+            # well. So, we only handle the args which type and value are exactly match
+            # the DUMMY_TYPE.
+            # TODO(fyrestone): Maybe we can remove the DUMMY_TYPE or make the
+            # DUMMY_TYPE None.
```

Comment:
Can you add a comment 

```
# https://github.com/ray-project/ray/pull/32478/files
```
here so that people would know why we do this?

Reply:
OK